src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/BKTree.Serialization.cs

```diff
@@ -3,46 +3,28 @@
 // See the LICENSE file in the project root for more information.
 
 using Microsoft.CodeAnalysis.Internal.Log;
-using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Serialization;
+using Microsoft.CodeAnalysis.Shared.Extensions;
 
 namespace Roslyn.Utilities;
 
 internal readonly partial struct BKTree
 {
     internal void WriteTo(ObjectWriter writer)
     {
-        writer.WriteInt32(_concatenatedLowerCaseWords.Length);
-        foreach (var c in _concatenatedLowerCaseWords)
-            writer.WriteChar(c);
```

Comment:
instead of writing out a char at a time, we can just write out the entire char[] (objectreader/writer supports that).